Domain/Shipping/Cargo/Itinerary.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Domain.Shipping.Location;
+using Domain.Shipping.Voyage;
+
+namespace Domain.Shipping.Cargo
+{
+    public class Itinerary 
+    {
+        public IReadOnlyCollection<Leg> Legs { get; }
+
+        public Itinerary(IList<Leg> legs)
+        {
+            if (legs == null)
+                throw new ArgumentNullException(nameof(legs));
+            if (legs.Count == 0)
+                throw new ArgumentException("legs cannot be empty");
+
+            Legs = new ReadOnlyCollection<Leg>(legs);
+        }
+
+        public UnLocode FirstLoadLocation => Legs.First().LoadLocation;
```

Comment:
If `Legs` is empty then what?

Reply:
Legs are not going to be empty it's an invariant checking that on the constructor a few lines above the statement.

Reply:
We could try declaring them (later on, in F#) as `NonEmptyList<Leg>`.